### DISCUSSION TITLE
Deprecate `String.is_valid_identifier()`

### DISCUSSION THREAD

**Discussion by timothyqiu** [godotengine/godot-proposals#9851] [Engine Core]
`String.is_valid_identifier()` currently checks if the string is ascii identifier (`0-9A-Za-z_`, not starting with a digit). This does not match the current rule of GDScript identifier anymore.

I suggest

- Adding `is_valid_ascii_identifier()`
- Adding `is_valid_unicode_identifier()`
- Deprecating `is_valid_identifier()`, making it an alias of `is_valid_ascii_identifier()` for compatibility reason.

**Comment by akien-mga** [comment]
Implemented by https://github.com/godotengine/godot/pull/95984.
